velox/functions/sparksql/tests/SparkCastExprTest.cpp

```diff
@@ -453,6 +454,36 @@ TEST_F(SparkCastExprTest, overflow) {
   testCast(
       makeNullableFlatVector<int64_t>({214748364890}, DECIMAL(12, 2)),
       makeNullableFlatVector<int64_t>({2147483648}));
+
+  VELOX_ASSERT_THROW(
+      (testCast<std::string, int8_t>("tinyint", {"166"}, {std::nullopt})),
+      "Cannot cast VARCHAR '166' to TINYINT. Value is too large for type.");
+  testTryCast<std::string, int8_t>("tinyint", {"166"}, {std::nullopt});
```

Comment:
Maybe we don't need to test the try cast, because it tests framework not this bug fix.